src/Microsoft.Health.Fhir.Core/Features/Search/SearchOptions.cs

```diff
@@ -97,6 +97,16 @@ internal set
         /// </summary>
         public IReadOnlyList<(SearchParameterInfo searchParameterInfo, SortOrder sortOrder)> Sort { get; internal set; }
 
+        /// <summary>
+        /// Marks whether we need to execute the second set of queries for (certain types of) sort.
+        /// </summary>
+        public bool SortQuerySecondPhase { get; internal set; } = false;
+
+        /// <summary>
+        /// Sets whether this search query is of type sort with filter.
+        /// </summary>
+        public bool IsSortWithFilter { get; internal set; } = false;
+
```

Comment:
Can these information be passed in a different way without having to be part of the SearchOptions class?

Reply:
Since part of the access for these properties is in the code which is data layer agnostic, I think this is the correct location for these properties.